### DISCUSSION TITLE
flet.Column is not aligning its child controls

### DISCUSSION THREAD

**Discussion by thetrotfreak** [flet-dev/flet#1093] [Q&A] [answered]
### Question

When returning a `flet.Column()` layout control from the `build()` method of a class inheriting from `flet.UserControl`, the `alignment` and `horizontal_alignment` parameters are not affecting the alignment  of its (`flet.Column.controls[]`) child controls. `flet.Row()` works fine.

### Code sample

```python
# row.py
from flet import Row, Text, UserControl, MainAxisAlignment


class RowControl(UserControl):
    def __init__(self):
        super().__init__()

    def build(self):
        return Row(
            controls=[Text(value="Hello World!")],
            alignment=MainAxisAlignment.CENTER
        )

# column.py
from flet import Column, UserControl, MainAxisAlignment
from row import RowControl


class ColumnControl(UserControl):
    def __init__(self):
        super().__init__()

    def build(self):
        return Column(
            controls=[RowControl(), RowControl()],
            alignment=MainAxisAlignment.SPACE_BETWEEN,
        )

# main.py
import flet
from column import ColumnControl


def main(page: flet.Page):
    page.add(ColumnControl())


if __name__ == "__main__":
    flet.app(target=main)
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by iron3oxide** [comment]
I just fiddled around a bit and it seems to me that using Column or Row directly doesn't change the behaviour. With Row, you can change the horizontal alignment, but vertical_alignment doesn't really do anything in basic examples. With Column, neither alignment visibly changes anything in those examples. Strange that I never noticed this before..

**Reply by thetrotfreak** [reply]
Yea, using `Column` and `Row` directly works as it should. It's just that only the `Column` seems to have some problem when it's returned from a `UserControl`.

**Reply by iron3oxide** [reply]
That's not what I meant, in the scenarios I tested it made no difference whether the Column was inside a UserControl or not. Could you maybe show a code sample where they behave differently?

**Reply by thetrotfreak** [reply]
Here's what I'm expecting from the original code, the 2 `Row` of `Text` **Hello World!** & **Hello World!** should be _center aligned_ on the page and that one of 'em should be on the _top_ while the other on the _bottom_ (because of `MainAxisAlignment.SPACE_BETWEEN`).

**Reply by iron3oxide** [reply]
I agree that the result is somewhat unexpected, but I just tried it again and no matter if I use ColumnControl or Column directly, the result is the same. So if anything, it's a problem of the controls themselves, not of using them in a UserControl.

**Reply by thetrotfreak** [reply]
Here's my [repo](https://github.com/thetrotfreak/33), if you could have a look.

![Screenshot_20230226_180703.png](https://user-images.githubusercontent.com/45330487/221410964-84d252e0-17e4-40b0-878c-7ae69e3db3c1.png)

I want the Scramble, Timer, & Buttons to be spaced in between, not cramped in together.

**Comment by thetrotfreak** [comment, marked as answer]
Read [this](https://github.com/flet-dev/flet/issues/1117#issuecomment-1453844094) discussion thread for answer.
